pkg/cmd/join/exec.go

```diff
@@ -213,84 +237,129 @@ func (o *Options) validate() error {
 }
 
 func (o *Options) run() error {
-	_, apiExtensionsClient, _, err := helpers.GetClients(o.ClusteradmFlags.KubectlFactory)
+	kubeClient, apiExtensionsClient, _, err := helpers.GetClients(o.ClusteradmFlags.KubectlFactory)
 	if err != nil {
 		return err
 	}
 
-	available, err := checkIfRegistrationOperatorAvailable(o.ClusteradmFlags.KubectlFactory)
-	if err != nil {
-		return err
-	}
+	r := reader.NewResourceReader(o.builder, o.ClusteradmFlags.DryRun, o.Streams)
 
-	files := []string{}
-	// If Deployment/klusterlet is not deployed, deploy it
-	if !available {
-		files = append(files,
-			"join/klusterlets.crd.yaml",
-			"join/namespace.yaml",
-			"join/service_account.yaml",
-			"join/cluster_role.yaml",
-			"join/cluster_role_binding.yaml",
-		)
-	}
-	files = append(files,
-		"join/namespace_agent.yaml",
-		"join/bootstrap_hub_kubeconfig.yaml",
-	)
+	if o.standalone {
```

Comment:
I'd we put standalone deploy in a separated functions and return fast. this big if-else section could be confusing